internal/procedures/server/start.go

```diff
@@ -74,3 +78,57 @@ func startCourse(course *model.Course) {
 		log.Error("Failed to update course.", err, course)
 	}
 }
+
+// Check to see if a server is running and start one if it's not.
+// Returns (false, 0) if a the primary server or cmd test server is already running
+// or (true, oldPort) if it started it's own server.
```

Comment:
This function (and comment) is missing a case.
What happens if a CMD_SERVER is running?

Reply:
If a CMD_SERVER is running, another CMD_SERVER will not be able to start since it will get stopped from starting when it reads the status file and finds out the pid in the status file is still active (isAlive()). Should I add this as a comment even though this function isn't the one doing that checking?

Reply:
You are relying on other code to set some external state and hopefully this function never gets called in that case.
You can see how this is pretty fragile.
Specifically, you are not [programming defensively](https://en.wikipedia.org/wiki/Defensive_programming), which you should generally be doing.
When programming defensively with enums, that usually means ensuring that all cases are accounted for.